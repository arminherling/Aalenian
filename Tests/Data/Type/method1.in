type Zero
{
    def value()
    {
        zero = 0
        return zero
    }
}
