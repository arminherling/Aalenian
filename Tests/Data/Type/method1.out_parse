TypeDefinitionStatement: {
    NameExpression: Zero
    BlockNode(1): {
        MethodDefinitionStatement: {
            NameExpression: value
            ParametersNode(0): {
            }
            BlockNode(2): {
                AssignmentStatement: {
                    Left: {
                        NameExpression: zero
                    }
                    Right: {
                        NumberLiteral: 0
                    }
                }
                ReturnStatement: {
                    NameExpression: zero
                }
            }
        }
    }
}
