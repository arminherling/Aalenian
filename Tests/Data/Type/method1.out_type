TypedTypeDefinitionStatement: {
    TypeKind: type
    TypeName: Zero
    Fields(0): {
    }
    Methods(1): {
        TypedMethodDefinitionStatement: {
            TypeKind: method
            Type: (ref Zero)->(i32)
            Name: value
            Parameters(1): {
                Parameter: {
                    Type: ref Zero
                    Name: 
                }
            }
            ReturnType: {
                Type: i32
            }
            Body(2): {
                TypedAssignmentStatement: {
                    Type: i32
                    Left: {
                        TypedVariable: {
                            Type: i32
                            Name: zero
                        }
                    }
                    Right: {
                        I32Value: {
                            Type: i32
                            Value: 0
                        }
                    }
                }
                TypedReturnStatement: {
                    Type: i32
                    Expression: {
                        TypedVariable: {
                            Type: i32
                            Name: zero
                        }
                    }
                }
            }
        }
    }
}
