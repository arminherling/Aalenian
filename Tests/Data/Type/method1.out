TypeDefinitionStatement: {
    Name: Zero
    Block(1): {
        MethodDefinitionStatement: {
            Name: value
            Parameters(0): {
            }
            Block(2): {
                AssignmentStatement: {
                    Left: {
                        Name: zero
                    }
                    Right: {
                        Number: 0
                    }
                }
                ReturnStatement: {
                    Name: zero
                }
            }
        }
    }
}
